crates/primitives/src/chain/spec.rs

```diff
@@ -1053,13 +1053,21 @@ impl From<Genesis> for ChainSpec {
 
         hardforks.extend(time_hardforks);
 
+        // NOTE: we prune receipts in full nodes, but not the deposit contract receipts. We do not
+        // have the deployment block in the genesis file, so we use block zero. We use the same
+        // deposit topic as the mainnet contract if we have the deposit contract address in the
+        // genesis json.
+        let deposit_contract = genesis.config.deposit_contract_address.map(|address| {
+            DepositContract { address, block: 0, topic: MAINNET_DEPOSIT_CONTRACT.topic }
+        });
```

Comment:
seems we only use the address, if so, we should not create a `DepositContract` object, just store the addr?

Reply:
we also use it in pruning, so we should store more than just an address